### DISCUSSION TITLE
Could you please help me in resolving the build error from "openbmc/build/ast2500-default$ bitbake obmc-phosphor-image"

### DISCUSSION THREAD

**Discussion by wangliuusa** [community/community#57267] [Programming Help] [answered]
### Select Topic Area

Show & Tell

### Select Feature Area

- [X] Issues
- [X] Projects

### Body

Sir / Madam,

Could you please help me in resolving the build error from "openbmc/build/ast2500-default$ bitbake obmc-phosphor-image ?"

Screen shot cut-pasted below:

Sincerely,
Liu Wang

liuw@liuw-SYS-E100-12T-H:~$ cd openbmc
liuw@liuw-SYS-E100-12T-H:~/openbmc$ ls -a
.               .gitreview   meta-aspeed-sdk  meta-google         meta-nuvoton       meta-raspberrypi  oe-init-build-env  .templateconf
..              LICENSE      meta-asrock      meta-hpe            meta-openembedded  meta-security     openbmc-env
bitbake         meta         meta-bytedance   meta-ibm            meta-openpower     meta-skeleton     OWNERS
.eslintrc.json  meta-amd     meta-delta       meta-ingrasys       meta-phosphor      meta-supermicro   poky
.git            meta-ampere  meta-evb         meta-inspur         meta-poky          meta-tyan         README.md
.github         meta-arm     meta-facebook    meta-intel-openbmc  meta-qualcomm      meta-wistron      scripts
.gitignore      meta-aspeed  meta-fii         meta-inventec       meta-quanta        meta-yadro        setup
liuw@liuw-SYS-E100-12T-H:~/openbmc$ git checkout v08.05
Note: switching to 'v08.05'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 064b33df2 Merge remote-tracking branch 'upstream/aspeed-master' into aspeed-master
liuw@liuw-SYS-E100-12T-H:~/openbmc$ source setup ast2500-default
Machine ast2500-default found in meta-aspeed-sdk/meta-ast2500-sdk
You had no conf/local.conf file. This configuration file has therefore been
created for you from /home/liuw/openbmc/meta-aspeed-sdk/meta-ast2500-sdk/conf/templates/default/local.conf.sample
You may wish to edit it to, for example, select a different MACHINE (target
hardware).

You had no conf/bblayers.conf file. This configuration file has therefore been
created for you from /home/liuw/openbmc/meta-aspeed-sdk/meta-ast2500-sdk/conf/templates/default/bblayers.conf.sample
To add additional metadata layers into your configuration please add entries
to conf/bblayers.conf.

The Yocto Project has extensive documentation about OE including a reference
manual which can be found at:
    https://docs.yoctoproject.org

For more information about OpenEmbedded see the website:
    https://www.openembedded.org/

Common targets are:
     # OpenBMC Firmware
     obmc-phosphor-image
liuw@liuw-SYS-E100-12T-H:~/openbmc/build/ast2500-default$ source setup $MACHINE $BUILD_DIR
bash: setup: No such file or directory
liuw@liuw-SYS-E100-12T-H:~/openbmc/build/ast2500-default$ source setup ast2500-default  $BUILD_DIR
bash: setup: No such file or directory
liuw@liuw-SYS-E100-12T-H:~/openbmc/build/ast2500-default$ bitbake obmc-phosphor-image
Loading cache: 100% |                                                                                                           | ETA:  --:--:--
Loaded 0 entries from dependency cache.
WARNING: /home/liuw/openbmc/meta-phosphor/recipes-phosphor/mrw/mrw-api-native.bb: QA Issue: mrw-api-native: native/nativesdk class is not inherited last, this can result in unexpected behaviour. Classes inherited after native/nativesdk: mrw-rev.bbclass perl-version.bbclass cpan-base.bbclass perlnative.bbclass [native-last]
Parsing recipes: 100% |##########################################################################################################| Time: 0:00:51
Parsing of 2706 .bb files complete (0 cached, 2706 parsed). 4200 targets, 395 skipped, 0 masked, 0 errors.
WARNING: No bb files in default matched BBFILE_PATTERN_ast2500-sdk-layer '^/home/liuw/openbmc/meta-aspeed-sdk/meta-ast2500-sdk/'
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "2.2.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-22.04"
TARGET_SYS           = "arm-openbmc-linux-gnueabi"
MACHINE              = "ast2500-default"
DISTRO               = "openbmc-phosphor"
DISTRO_VERSION       = "nodistro.0"
TUNE_FEATURES        = "arm thumb arm1176jzs"
TARGET_FPU           = "soft"
meta                 
meta-poky            
meta-oe              
meta-networking      
meta-python          
meta-phosphor        
meta-aspeed-sdk      
meta-ast2500-sdk     = "HEAD:064b33df2b63fb2a4f93859499380161744addc3"

Initialising tasks: 100% |#######################################################################################################| Time: 0:00:01
Sstate summary: Wanted 1997 Local 0 Mirrors 0 Missed 1997 Current 0 (0% match, 0% complete)
NOTE: Executing Tasks
WARNING: cracklib-2.9.8-r0 do_fetch: Failed to fetch URL git://github.com/cracklib/cracklib;protocol=https;branch=master, attempting MIRRORS if available
WARNING: ncurses-native-6.3+20220423-r0 do_fetch: Failed to fetch URL git://github.com/mirror/ncurses.git;protocol=https;branch=master, attempting MIRRORS if available
Setscene tasks: 1997 of 1997
Currently  8 running tasks (1423 of 4830)  29% |###########################                                                                    |
0: linux-aspeed-5.15.0+gitAUTOINC+d1238fd5b5-r0 do_fetch - 41m3s (pid 370272)  31% |##############                                | 271.00 KiB/s
Setscene tasks: 1997 of 1997
Currently  8 running tasks (1507 of 4830)  31% |#############################                                                                  |
0: linux-aspeed-5.15.0+gitAUTOINC+d1238fd5b5-r0 do_fetch - 46m16s (pid 370272)  32% |##############                               | 364.00 KiB/s
1: fmt-9.1.0-r0 do_fetch - 15m30s (pid 1036564)  47% |###############################################################                                                                         | 14.00 KiB/s
WARNING: libpcre-8.45-r0 do_fetch: Failed to fetch URL https://downloads.sourceforge.net/pcre/pcre-8.45.tar.bz2, attempting MIRRORS if available
WARNING: fmt-9.1.0-r0 do_fetch: Failed to fetch URL git://github.com/fmtlib/fmt;branch=master;protocol=https, attempting MIRRORS if available
WARNING: avahi-0.8-r0 do_fetch: Failed to fetch URL https://github.com/lathiat/avahi/releases//download/v0.8/avahi-0.8.tar.gz, attempting MIRRORS if available
WARNING: c-ares-native-1.18.1-r0 do_fetch: Failed to fetch URL git://github.com/c-ares/c-ares.git;branch=main;protocol=https, attempting MIRRORS if available
WARNING: ipmitool-1.8.18+gitAUTOINC+c3939dac2c-r0 do_fetch: Failed to fetch URL git://github.com/ipmitool/ipmitool.git;protocol=https;branch=master, attempting MIRRORS if available
ERROR: ipmitool-1.8.18+gitAUTOINC+c3939dac2c-r0 do_fetch: Fetcher failure: Fetch command export PSEUDO_DISABLED=1; export DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/1000/bus"; export SSH_AUTH_SOCK="/run/user/1000/keyring/ssh"; export PATH="/home/liuw/openbmc/scripts:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot-native/usr/bin/arm-openbmc-linux-gnueabi:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot/usr/bin/crossscripts:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot-native/usr/sbin:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot-native/usr/bin:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot-native/sbin:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot-native/bin:/home/liuw/openbmc/poky/bitbake/bin:/home/liuw/openbmc/build/ast2500-default/tmp/hosttools"; export HOME="/home/liuw"; LANG=C git -c gc.autoDetach=false -c core.pager=cat clone --bare --mirror https://github.com/ipmitool/ipmitool.git /home/liuw/openbmc/build/ast2500-default/downloads/git2/github.com.ipmitool.ipmitool.git --progress failed with exit code 128, no output
ERROR: ipmitool-1.8.18+gitAUTOINC+c3939dac2c-r0 do_fetch: Bitbake Fetcher Error: FetchError('Unable to fetch URL from any source.', 'git://github.com/ipmitool/ipmitool.git;protocol=https;branch=master')
ERROR: Logfile of failure stored in: /home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/temp/log.do_fetch.1415947
ERROR: Task (/home/liuw/openbmc/meta-openembedded/meta-oe/recipes-kernel/ipmitool/ipmitool_1.8.19.bb:do_fetch) failed with exit code '1'
WARNING: phosphor-debug-collector-1.0+gitAUTOINC+4cb0799f16-r1 do_fetch: Failed to fetch URL git://github.com/openbmc/phosphor-debug-collector;branch=master;protocol=https, attempting MIRRORS if available
ERROR: phosphor-debug-collector-1.0+gitAUTOINC+4cb0799f16-r1 do_fetch: Fetcher failure: Fetch command export PSEUDO_DISABLED=1; export DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/1000/bus"; export SSH_AUTH_SOCK="/run/user/1000/keyring/ssh"; export PATH="/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/phosphor-debug-collector/1.0+gitAUTOINC+4cb0799f16-r1/recipe-sysroot-native/usr/bin/python3-native:/home/liuw/openbmc/scripts:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/phosphor-debug-collector/1.0+gitAUTOINC+4cb0799f16-r1/recipe-sysroot-native/usr/bin/arm-openbmc-linux-gnueabi:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/phosphor-debug-collector/1.0+gitAUTOINC+4cb0799f16-r1/recipe-sysroot/usr/bin/crossscripts:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/phosphor-debug-collector/1.0+gitAUTOINC+4cb0799f16-r1/recipe-sysroot-native/usr/sbin:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/phosphor-debug-collector/1.0+gitAUTOINC+4cb0799f16-r1/recipe-sysroot-native/usr/bin:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/phosphor-debug-collector/1.0+gitAUTOINC+4cb0799f16-r1/recipe-sysroot-native/sbin:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/phosphor-debug-collector/1.0+gitAUTOINC+4cb0799f16-r1/recipe-sysroot-native/bin:/home/liuw/openbmc/poky/bitbake/bin:/home/liuw/openbmc/build/ast2500-default/tmp/hosttools"; export HOME="/home/liuw"; LANG=C git -c gc.autoDetach=false -c core.pager=cat clone --bare --mirror https://github.com/openbmc/phosphor-debug-collector /home/liuw/openbmc/build/ast2500-default/downloads/git2/github.com.openbmc.phosphor-debug-collector --progress failed with exit code 128, no output
ERROR: phosphor-debug-collector-1.0+gitAUTOINC+4cb0799f16-r1 do_fetch: Bitbake Fetcher Error: FetchError('Unable to fetch URL from any source.', 'git://github.com/openbmc/phosphor-debug-collector;branch=master;protocol=https')
ERROR: Logfile of failure stored in: /home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/phosphor-debug-collector/1.0+gitAUTOINC+4cb0799f16-r1/temp/log.do_fetch.1363696
ERROR: Task (/home/liuw/openbmc/meta-phosphor/recipes-phosphor/dump/phosphor-debug-collector_git.bb:do_fetch) failed with exit code '1'
NOTE: Tasks Summary: Attempted 2271 tasks of which 0 didn't need to be rerun and 2 failed.

Summary: 2 tasks failed:
  /home/liuw/openbmc/meta-openembedded/meta-oe/recipes-kernel/ipmitool/ipmitool_1.8.19.bb:do_fetch
  /home/liuw/openbmc/meta-phosphor/recipes-phosphor/dump/phosphor-debug-collector_git.bb:do_fetch
Summary: There were 10 WARNING messages.
Summary: There were 4 ERROR messages, returning a non-zero exit code.
liuw@liuw-SYS-E100-12T-H:~/openbmc/build/ast2500-default$ 

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @wangliuusa ! 

You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects.We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion. 

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by RamtinKosari** [comment, marked as answer]
Hi dear @wangliuusa 
The build error you encountered is related to fetching the repositories "ipmitool" and "phosphor-debug-collector" during the build process. Here are the specific errors for each repository:
1. For ipmitool:
```awk
ERROR: ipmitool-1.8.18+gitAUTOINC+c3939dac2c-r0 do_fetch: Fetcher failure: Fetch command export PSEUDO_DISABLED=1; export DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/1000/bus"; export SSH_AUTH_SOCK="/run/user/1000/keyring/ssh"; export PATH="/home/liuw/openbmc/scripts:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot-native/usr/bin/arm-openbmc-linux-gnueabi:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot/usr/bin/crossscripts:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot-native/usr/sbin:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot-native/usr/bin:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot-native/sbin:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/recipe-sysroot-native/bin:/home/liuw/openbmc/poky/bitbake/bin:/home/liuw/openbmc/build/ast2500-default/tmp/hosttools"; export HOME="/home/liuw"; LANG=C git -c gc.autoDetach=false -c core.pager=cat clone --bare --mirror https://github.com/ipmitool/ipmitool.git /home/liuw/openbmc/build/ast2500-default/downloads/git2/github.com.ipmitool.ipmitool.git --progress failed with exit code 128, no output
ERROR: ipmitool-1.8.18+gitAUTOINC+c3939dac2c-r0 do_fetch: Bitbake Fetcher Error: FetchError('Unable to fetch URL from any source.', 'git://github.com/ipmitool/ipmitool.git;protocol=https;branch=master')
ERROR: Logfile of failure stored in: /home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/ipmitool/1.8.18+gitAUTOINC+c3939dac2c-r0/temp/log.do_fetch.1415947
ERROR: Task (/home/liuw/openbmc/meta-openembedded/meta-oe/recipes-kernel/ipmitool/ipmitool_1.8.19.bb:do_fetch) failed with exit code '1'
```
2. For phosphor-debug-collector:
```awk
ERROR: phosphor-debug-collector-1.0+gitAUTOINC+4cb0799f16-r1 do_fetch: Fetcher failure: Fetch command export PSEUDO_DISABLED=1; export DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/1000/bus"; export SSH_AUTH_SOCK="/run/user/1000/keyring/ssh"; export PATH="/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/phosphor-debug-collector/1.0+gitAUTOINC+4cb0799f16-r1/recipe-sysroot-native/usr/bin/python3-native:/home/liuw/openbmc/scripts:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabi/phosphor-debug-collector/1.0+gitAUTOINC+4cb0799f16-r1/recipe-sysroot-native/usr/bin/arm-openbmc-linux-gnueabi:/home/liuw/openbmc/build/ast2500-default/tmp/work/arm1176jzs-openbmc-linux-gnueabiIt seems that the build process is failing to fetch the repositories "ipmitool" and "phosphor-debug-collector" from their respective sources. The error message indicates that the Git fetch command failed with exit code 128, and there is no output available to provide further details about the error.
```
Here are a few possible reasons and solutions for the fetch failure:

1. Network connectivity: Check your network connection to ensure that you can access the internet and that there are no restrictions or firewalls blocking the Git fetch operation. You can try running a simple git clone command manually outside of the build process to see if you can fetch the repositories directly.

2. Repository availability: Verify that the repositories "ipmitool" and "phosphor-debug-collector" exist and are accessible from the specified sources. It's possible that the repositories might have been moved or deleted. Visit the repository URLs ([https://github.com/ipmitool/ipmitool.git ↗](https://github.com/ipmitool/ipmitool.git) and the relevant phosphor-debug-collector repository) in a web browser to confirm their availability.

3. Proxy settings: If you are behind a proxy, ensure that your build environment is properly configured to use the proxy. Check if there are any environment variables (http_proxy, https_proxy, git_proxy) that need to be set to the correct proxy URL.

4. Fetch protocol: The error message mentions using the https protocol for fetching the repositories. However, it's also possible to try using the git protocol instead. You can modify the recipe or configuration files to use the git protocol instead of https.

5. Repository access permissions: Ensure that you have the necessary permissions to access the repositories. If they are private repositories, make sure you have the appropriate credentials (username/password, SSH key) configured in your build environment to authenticate and access the repositories.

6. Repository branches: Verify that the specified branches (master in the error message) exist in the repositories. If the branches have been renamed or removed, you may need to update the branch names in your build configuration.

By investigating these possible causes and applying the corresponding solutions, you should be able to resolve the fetch errors and successfully build your project.

**Reply by wangliuusa** [reply]
HI [RamtinKosari](https://github.com/RamtinKosari),

Our IT has removed the blocking to the url for fetching the necessities and it is building now.

Sincerely,
Liu
